r/src/expression.cpp

```diff
@@ -20,82 +20,72 @@
 #if defined(ARROW_R_WITH_ARROW)
 
 // [[arrow::export]]
-std::shared_ptr<arrow::dataset::FieldExpression> dataset___expr__field_ref(
-    std::string name) {
-  return std::make_shared<arrow::dataset::FieldExpression>(std::move(name));
+ds::ExpressionPtr dataset___expr__field_ref(std::string name) {
+  return std::make_shared<ds::FieldExpression>(std::move(name));
 }
 
 // [[arrow::export]]
-std::shared_ptr<arrow::dataset::ComparisonExpression> dataset___expr__equal(
-    const std::shared_ptr<arrow::dataset::Expression>& lhs,
-    const std::shared_ptr<arrow::dataset::Expression>& rhs) {
-  return arrow::dataset::equal(lhs, rhs);
+ds::ExpressionPtr dataset___expr__equal(const ds::ExpressionPtr& lhs,
+                                        const ds::ExpressionPtr& rhs) {
+  return ds::equal(lhs, rhs);
 }
 
 // [[arrow::export]]
-std::shared_ptr<arrow::dataset::ComparisonExpression> dataset___expr__not_equal(
-    const std::shared_ptr<arrow::dataset::Expression>& lhs,
-    const std::shared_ptr<arrow::dataset::Expression>& rhs) {
+ds::ExpressionPtr dataset___expr__not_equal(const ds::ExpressionPtr& lhs,
+                                            const ds::ExpressionPtr& rhs) {
   return arrow::dataset::not_equal(lhs, rhs);
```

Comment:
```suggestion
  return ds::not_equal(lhs, rhs);
```
and same for the rest, right?